src/Groups/homomorphisms.jl

```diff
@@ -568,6 +576,57 @@ function isomorphism(::Type{FPGroup}, G::GAPGroup; on_gens::Bool=false)
    end::GAPGroupHomomorphism{typeof(G), FPGroup}
 end
 
+# If `G` is not a full pc group then switch to a full pc group in Oscar.
+function isomorphism(T::Type{PcGroup}, G::GAPGroup; on_gens::Bool=false)
+   # Known isomorphisms are cached in the attribute `:isomorphisms`.
+   isos = get_attribute!(Dict{Tuple{Type, Bool}, Any}, G, :isomorphisms)::Dict{Tuple{Type, Bool}, Any}
+   return get!(isos, (T, false)) do
+     if on_gens
+       # Throw an exception if `gens(G)` is not a polycyclic sequence for `G`.
+       # Create a group in `IsPcGroup` if `G` is finite,
+       # and a group in `IsPcpGroup` otherwise.
+       if is_finite(G)
+         fam = GAP.Globals.ElementsFamily(GAP.Globals.FamilyObj(G.X))::GapObj
+         Ggens = GapObj([x.X for x in gens(G)])::GapObj
```

Comment:
Huh, does this line work as intended? There is no test case covering this code according to Codecov, and I have some doubts: won't this create a GAP list of OSCAR elements?

Perhaps like this:
```suggestion
         Ggens = GapObj(gens(G); recursive=true)
```

Reply:
`gens(G)` consists of OSCAR elements, the array of the `x.X` consists of GAP elements.
I am going to add tests.
(And yes, the code you propose is better.)